force-app/main/default/classes/SOQL.cls

```diff
@@ -130,9 +184,13 @@ public with sharing class SOQL {
      * @return A list of Lead records.
      * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}    
+    public static List<Lead> getLeadsBySourceOrTitle (String leadSource, String keyword) {
+        List<Lead> leadsBySourceOrTitle = [SELECT Id, LastName, Company, Phone, Email, LeadSource, Title
+                                            FROM Lead
+                                            WHERE LeadSource = :leadSource OR Title LIKE :('%' + keyword + '%')];
+
+        return leadsBySourceOrTitle;    
```

Comment:
I like the formatting here, but I would put the fields you select oriented vertically. All great.